robottelo/hosts.py

```diff
@@ -479,6 +479,118 @@ def install_cockpit(self):
         if result.status != 0:
             raise ContentHostError('Failed to install the cockpit')
 
+    def register(
+        self,
+        target=settings.server.hostname,
+        org='Default_Organization',
+        loc='Default_Location',
+        activation_keys=None,
+        setup_insights=False,
+        setup_remote_execution=True,
+        setup_remote_execution_pull=False,
+        lifecycle_environment=None,
+        operating_system=None,
+        packages=None,
+        repo=None,
+        repo_gpg_key_url=None,
+        remote_execution_interface=None,
+        update_packages=False,
+        ignore_subman_errors=True,
+        force=True,
+        insecure=True,
+        username=settings.server.admin_username,
+        password=settings.server.admin_password,
+    ):
+        """Registers content host to the Satellite or Capsule server
+        using a global registration template.
+
+        :param target: Satellite or Capusle hostname to register to, required.
+        :param org: Organization name to register content host for, required.
+        :param loc: Location name to register content host for, required.
+        :param activation_keys: Activation key name to register content host
+            with, required.
+        :param setup_insights: Install and register Insights client, requires OS repo.
+        :param setup_remote_execution: Copy remote execution SSH key.
+        :param setup_remote_execution_pull: Deploy pull provider client on host
+        :param lifecycle_environment: Lifecycle environment.
+        :param operating_system: Operating system.
+        :param packages: A list of packages to install on the host when registered.
+        :param repo: Repository to be added before the registration is performed, supply url.
+        :param repo_gpg_key_url: Public key to verify the package signatures, supply url.
+        :param remote_execution_interface: Identifier of the host interface for remote execution.
+        :param update_packages: Update all packages on the host.
+        :param ignore_subman_errors: Ignore subscription manager errors.
+        :param force: Register the content host even if it's already registered.
+        :param insecure: Don't verify server authenticity.
+        :param username: A user name to register the content host with.
+        :param password: The user password.
+        :return: SSHCommandResult instance filled with the result of the
+            registration.
+        """
+        cmd = 'curl -sS '
+        if username and password:
+            cmd += f'-u {username}:{password} '
```

Comment:
I think this style of constructing the command is difficult to read and make sense of, and is using `+` to construct the string from f-string parts, instead of using f-string directly.

How about doing whatever parameter validation you need to do ahead of time, and then constructing the string in a more easy to read format.

```
user = target.username
pass = target.password

cmd = (
    f'curl -sS '
    f'{"--insecure " if insecure else ""}'
    f'https://{target.hostname}:{target.port}/register?'
    f'activation_keys={activation_keys}'
    # etc
)
```

Reply:
Yeah sometimes you have to mix things a bit, and label some variables that only end up getting used once so that the final string construction is readable. Taking a look at the new commit(s) now, thanks for revising!